app.js

```diff
@@ -143,17 +143,17 @@ function _isEqual(arrayOne, arrayTwo) {
     : false;
 }
 
-function _lastIndexOf(array, element) {
-  for (let i = array.length - 1; i >= 0; i--) {
-    if (array[i] === element) {
+Array.prototype._lastIndexOf = function(element) {
+  for (let i = this.length - 1; i >= 0; i--) {
+    if (this[i] === element) {
       return i;
     }
   }
   return -1;
 }
 
-function _concat(...arrays) {
-  return _reduce(
+Array.prototype._concat = function(...arrays) {
```

Comment:
Un par de cosas:
1. El arrays dentro del _reduce sobra, porque también es el array sobre el que hay que aplicar la función. O sea, se aplica el _reduce sobre el array de arrays que se pasa como argumento. Y...
2. El this, hay que pasarlo como valor inicial en lugar del array vacío. O sea
```
Array.prototype._concat = function(...arrays) {
  return arrays._reduce(
    (concatenatedArray, currentArray) => {
      for (element of currentArray) {
        concatenatedArray.push(element);
      }
      return concatenatedArray;
    },
    this
  );
}
```